rbac/server/db/users_query.py

```diff
@@ -208,20 +208,33 @@ def fetch_user_ids_by_manager(remote_id):
             .coerce_to("array")
             .run(conn)
         )
-        if "manager_id" in user_object[0]:
-            if user_object[0]["manager_id"]:
-                manager_id = user_object[0]["manager_id"]
+        if user_object[0]:
+            manager_id = user_object[0]["manager_id"]
+            manager_object = await (
+                r.db("rbac")
+                .table("users")
+                .filter({"remote_id": manager_id})
+                .coerce_to("array")
+                .run(conn)
+            )
+            if manager_object:
+                if manager_object[0]:
+                    manager_chain.append(manager_object[0]["next_id"])
+                    next_id = manager_object[0]["next_id"]
+            else:
                 manager_object = await (
```

Comment:
This looks like it's duplicating lines 212-224, you might want to break this off into its own function and pass in "next_id" or "remote_id", and return the manager's `next_id` since that looks like the only change between them.

Reply:
its not duplicating first one filters by remote_id and the second one filters by next id table attributes were different

Reply:
yes, but the two code blocks are identical except for that filter, right?

Reply:
its like if and else condition if first one finds the search it will not run the second block else it runs the second block for search

Reply:
I get that, but it still seems like there's a lot of duplication there. 
if you're getting the first user where the remote_id or next_id equals the manager_id, you might be able to get away with an `or` statement in your filter, https://rethinkdb.com/api/python/filter/#basic-predicates 

but I don't think we should ever have a remote_id stored in the manager_id field in the first place. I think we're only using next_id. @nadiabahrami can you confirm?

Reply:
I updated it please check it